osgar/drivers/logsocket.py

```diff
@@ -39,6 +39,9 @@ def join(self, timeout=None):
 
     def run_input(self):
         while self.bus.is_alive():
+            if self.pair == (None, None):  # unknown address
+                self.bus.sleep(0.1)
+                continue
```

Comment:
This should not be needed here. The input thread should not run if the socket is not connected yet. So I think starting the threads should be delegated to the specific classes from the parent.

Reply:
Good point, it should not be started. There is refactoring attempt available, where as plus I see that only `LogTCPDynamicIP` is not handling channel `addr` but several methods are semi-duplicated. I will add one more test for case when you receive `addr` twice.